src/routes/trainingSession/trainingSession.controller.ts

```diff
@@ -0,0 +1,71 @@
+import { TrainingSession } from "../../schemas/training.session";
+import { Request, Response } from 'express';
+import { TrainingSessionDocument, TTrainingSession } from "../../types/db/training.session.types";
+
+export async function getTrainingSession(
+    req: Request,
+    res: Response
+    ){
+        try {
+            let trainingSessionDocs: (TrainingSessionDocument & Required<{ _id: TTrainingSession; }>)[] = [];
+
+            if (req.query.userId == null) {
+                throw new Error("No userId provided!")
+            }
+
+            if ( Array.isArray(req.query.id) ) {
+                req.query.id.forEach(async id => {
+                    const singleSession = await TrainingSession.findById(id)
+                    .populate({
+                        path: 'executions',
+                        populate: {
+                            path: 'exercise',
+                            model: 'Exercise'
+                        }
+                    })
+                    if (singleSession) {
+                        trainingSessionDocs.push(singleSession);
+                        
+                    }
+                })
```

Comment:
```suggestion
                    const ids = id;
                    const singleSession = await TrainingSession.find({_id: {$in: ids})
                    .populate({
                        path: 'executions',
                        populate: {
                            path: 'exercise',
                            model: 'Exercise'
                        }
                    }) 
```

[https://kb.objectrocket.com/mongo-db/the-mongoose-in-operator-1015](https://kb.objectrocket.com/mongo-db/the-mongoose-in-operator-1015)